skfda/misc/operators/_linear_differential_operator.py

```diff
@@ -577,50 +579,188 @@ def bspline_penalty_matrix_optimized(
     return penalty_matrix
 
 
+def _optimized_operator_evaluation_in_grid(
+    linear_operator: LinearDifferentialOperator,
+    grid_points: NDArrayFloat,
+    findiff_accuracy: int,
+) -> Tuple[NDArrayFloat, NDArrayInt]:
+    """
+    Compute the linear operator applied to the delta basis of the grid.
+
+    The delta basis is the basis where the i-th basis function is 1 in the
+    i-th point and 0 in the rest of points.
+
+    Returns:
+     - A matrix where the i-th row contians the values of the linear operator
+        applied to the i-th basis function of the delta basis.
+
+    - The domain limits of each basis function of the delta basis. Outside
+        these limits, the result of applying the linear operator to the
+        given basis function is zero.
+
+    Example:
+        >>> import numpy as np
+        >>> from skfda.misc.operators import LinearDifferentialOperator
+        >>> _optimized_operator_evaluation_in_grid(
+        ...     LinearDifferentialOperator(2),
+        ...     np.linspace(0, 7, 8),
+        ...     2,
+        ... )
+        (array([[ 2.,  1.,  0.,  0.,  0.,  0.,  0.,  0.],
+        [-5., -2.,  1.,  0.,  0.,  0.,  0.,  0.],
+        [ 4.,  1., -2.,  1.,  0.,  0.,  0.,  0.],
+        [-1.,  0.,  1., -2.,  1.,  0.,  0.,  0.],
+        [ 0.,  0.,  0.,  1., -2.,  1.,  0., -1.],
+        [ 0.,  0.,  0.,  0.,  1., -2.,  1.,  4.],
+        [ 0.,  0.,  0.,  0.,  0.,  1., -2., -5.],
+        [ 0.,  0.,  0.,  0.,  0.,  0.,  1.,  2.]]),
+        array([[0, 1],
+                [0, 2],
+                [0, 3],
+                [0, 4],
+                [3, 7],
+                [4, 7],
+                [5, 7],
+                [6, 7]]))
+
+    Explanation:
+        Each row of the first array contains the values of the linear operator
+        applied to the delta basis function.
+        As it can be seen, each row of the second array cointains the domain
+        limits of the corresponding row of the first array. That is to say,
+        the range in which their values are not 0.
+        For example, the third row of the first array is non-zero until the
+        fourth element, so the third row of the second array is [0, 3].
+    """
+    n_points = grid_points.shape[0]
+    diff_coefficients = np.zeros((n_points, n_points))
+
+    domain_limits = np.hstack(
+        [
+            np.ones((n_points, 1)) * n_points,
+            np.zeros((n_points, 1)),
+        ],
+    ).astype(int)
+
+    # The desired matrix can be obtained by appending the coefficients
+    # of the finite differences for each point column-wise.
+    for dif_order, w in enumerate(linear_operator.weights):
```

Comment:
You should check `linear_operator.constant_weights` as the weights could be callables, in principle.